includes/demo.php

```diff
@@ -84,173 +84,129 @@ function wp_notify_add_admin_options_page() {
-				<label class="screen-reader-text" for="cb-select-123">SMS</label>
-				<input id="cb-select-123" type="checkbox" name="post[]" value="123">
-			</td>
-			<td scope="row" class="column-format">
-				<label class="screen-reader-text" for="cb-select-123">App</label>
-				<input id="cb-select-123" type="checkbox" name="post[]" value="123">
-			</td>
-		</tr>
-
-		</tbody>
-
-		<tfoot>
-		<tr>
-			<td id="cb" class="manage-column column-cb check-column"><label class="screen-reader-text" for="cb-select-all-1">Select All</label><input id="cb-select-all-1" type="checkbox"></td>
-			<th scope="col" id="source" class="manage-column column-source column-primary sortable desc"><a href="#"><span>Notification Source</span><span class="sorting-indicator"></span></a></th>
-			<td id="showin" class="manage-column column-format column-showin"><span>Show in:</span></td>
-			<th scope="col" id="admin" class="manage-column column-format column-admin desc"><span>Admin</span></th>
-			<th scope="col" id="email" class="manage-column column-format column-email desc"><span>Email</span></th>
-			<th scope="col" id="sms" class="manage-column column-format column-sms desc"><span>SMS</span></th>
-			<th scope="col" id="app" class="manage-column column-format column-app desc"><span>App</span></th>
-		</tr>
-		</tfoot>
-	</table>
-
 	<?php
+	/**
+	 * List_table that shows the settings
+	 */
+	class List_Table extends WP_List_Table {
+		/**
+		 * Prepare the items for the table to process
+		 */
+		public function prepare_items() {
+			$columns  = $this->get_columns();
+			$hidden   = $this->get_hidden_columns();
+			$sortable = $this->get_sortable_columns();
+
+			$data = $this->table_data();
+
+			$per_page     = 10;
+			$current_page = $this->get_pagenum();
+			$total_items  = count( $data );
+
+			$this->set_pagination_args(
+				array(
+					'total_items' => $total_items,
+					'per_page'    => $per_page,
+				)
+			);
+
+			$data = array_slice( $data, ( ( $current_page - 1 ) * $per_page ), $per_page );
+
+			$this->_column_headers = array( $columns, $hidden, $sortable );
+			$this->items           = $data;
+		}
+
+		/**
+		 * Override the parent columns method. Defines the columns to use in your listing table
+		 *
+		 * @return Array
+		 */
+		public function get_columns(): array {
+			return array(
+				'cb'     => '<input type="checkbox" />',
+				'source' => __( 'Notification Source' ),
+				'show'   => __( 'Show in:' ),
+				'admin'  => __( 'Admin' ),
+				'email'  => __( 'Email' ),
+				'sms'    => __( 'Sms' ),
+				'app'    => __( 'App' ),
+			);
+		}
+
+		public function get_hidden_columns(): array {
+			return array();
+		}
+
+		public function get_sortable_columns(): array {
+			return array( 'source' => array( 'source', false ) );
+		}
+
+		private function table_data(): array {
+			$data = array();
+
+			$indexes = array(
+				'WordPress',
+				'User activity (published, edited, ect)',
+				'Comments',
+				'Site health',
+			);
```

Comment:
With this change we can add items to 'who' emits the messages

```suggestion
			$indexes = apply_filters(
				'wp_feature_notifications_settings_indexes',
				array(
					'WordPress',
					'User activity (published, edited, ect)',
					'Comments',
					'Site health',
				)
			);
```

Reply:
> We'll need a way for plugins/themes to register or hook into this table to add their own rows- perhaps based on a notification source/category or else

In the same way we can provide a filter to expand or remove the 'way' you receive notifications. But perhaps for a slightly more complex operation this approach is not good enough and we will have to register the plugins that use the wp-notify. What do you think? Could this at least temporally provide a solution to the issue?

Reply:
I think this works well, at least in the short term. Longer term I think we'll have something more advanced set up.

Reply:
However, I should probably also add the filter docboc. correct?

Reply:
Ah very good point, I always forget that for filters.

Reply:
updated!